wal/wal.go

```diff
@@ -172,11 +172,8 @@ func openAtIndex(dirpath string, snap walpb.Snapshot, all bool) (*WAL, error) {
 		}
 		err = l.TryLock()
 		if err != nil {
-			if all {
+			if write {
 				return nil, err
-			} else {
```

Comment:
check the error is ErrLocked when skipping it in read mode, so it could know other errors early